src/view.c

```diff
@@ -850,4 +854,8 @@ view_destroy(struct view *view)
 	/* Remove view from server->views */
 	wl_list_remove(&view->link);
 	free(view);
+
+	if (need_cursor_update) {
+		cursor_update_focus(server, true);
```

Comment:
We do we need a `force_reenter` here?

Reply:
I was thinking we need it when exiting interactive mode but could be wrong. I'll double-check.

Reply:
`force_reenter` should only be required if we are already over a surface with active mouse focus but did change the cursor behind the back of the surface for some reason.

In this case the usual mouse focus stuff should just detect a different surface below the mouse (not owning cursor focus), thus send the enter event and let the new surface set the cursor image.

Reply:
With SSD resize margins, could we be resizing one surface (the one that gets destroyed here) while another is under the cursor and has mouse focus? That's the only case I can think of where it would matter one way or the other. If that's an impossible case, then I agree, the `force_reenter` here is redundant. It should also be harmless, so on the whole I would be happier keeping it, but go with whatever you think is best.

Reply:
I think once you move the mouse to reach the resize grabs the cursor focus is cleared (because there is no surface under the cursor). All the following movements are then handled by labwc until interactive_end() at which point the usual cursor focus stuff should kick back in, detect the new surface under the cursor and send enter -> application sets cursor -> business as usual.

At least that is how it works in my head, will verify this point when looking through this in more detail.

Reply:
I think I came up with a way we can avoid this kind of manual analysis and just let `cursor_update_common()` figure it out for us. See v2.

Reply:
Well done, I think that should work fine for us